### ISSUE TITLE
Owned type retrieval failure

### ISSUE THREAD

**Issue by ADringer** [dotnet/efcore#12429]
Hi,

I have a couple of classes as:

```C#
public class Donor
    {
        public string Id { get; set; }
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public bool ClaimGiftAid { get; set; }
        public Address Address { get; set; }
    }

public class Address
    {        
        public string Line1 { get; set; }
        public string Line2 { get; set; }
        public string Line3 { get; set; }
        public string Line4 { get; set; }
        public string Town { get; set; }
        public string County { get; set; }
        public string Postcode { get; set; }
        public CountryKind? Country { get; set; }
        public decimal? Latitude { get; set; }
        public decimal? Longitude { get; set; }
    }
```

and have set up the owned type as:
```C#
protected override void OnModelCreating(ModelBuilder modelBuilder)
        {
            base.OnModelCreating(modelBuilder);           
         modelBuilder.Entity<Donor>().OwnsOne(x => x.Address);
```

When attempting to retrieve a Donor I get the error:

> Invalid column name 'Country'

Looking in the logs I can see:

```
Optimized query model: 
'(from Donor x in DbSet<Donor>
join Address x.Address in DbSet<Address>
on Property([x], "Id") equals Property([x.Address], "DonorId") into x.Address_group
from Address x.Address in 
    (from Address x.Address_groupItem in [x.Address_group]
    select [x.Address_groupItem]).DefaultIfEmpty()
where [x].Id == __donorId_0
select Donor _Include(
    queryContext: queryContext, 
    entity: [x], 
    included: new object[]{ [x.Address] }, 
    fixup: (QueryContext queryContext | Donor entity | object[] included) => 
    {
        return !(bool ReferenceEquals(included[0], null)) ? 
        {
            return entity.Address = (Address)included[0]
        }
         : 
        {
            return default(Address)
        }

    }
)).First()'
```
```sql
SELECT TOP(1) [x].[Id], [x].[ClaimGiftAid], [x].[FirstName], [x].[LastName], [x.Address].[DonorId], [x.Address].[Country], [x.Address].[County], [x.Address].[Latitude], [x.Address].[Line1], [x.Address].[Line2], [x.Address].[Line3], [x.Address].[Line4], [x.Address].[Longitude], [x.Address].[Postcode], [x.Address].[Town]
FROM [Donor] AS [x]
LEFT JOIN [Address] AS [x.Address] ON [x].[Id] = [x.Address].[DonorId]
WHERE [x].[Id] = @__donorId_0
```
From the logs output it looks like it thinks `Address` is a different table and so doing a join to it when it's on the same table.

I am running this in an Azure Function, not sure if that will cause an issue?

### Further technical details
EF Core version: 2.1.1
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Environment: Azure Function V2

**Comment by smitpatel** [comment]
@ADringer - Is `Address` different table? Can you share what does table look like in database?

**Comment by ADringer** [comment]
@smitpatel It's all in one table, here's what the table looks like (there are more columns than the model defines)

![image](https://user-images.githubusercontent.com/19554490/41678560-43d59e1c-74c3-11e8-9205-9ec467db95e8.png)

**Comment by ADringer** [comment]
Sorry for the delay, had a sick child to look after!

I've worked out what my issue was. The DB was originally created using EF6 and I had configured it using `Remove<PluralizingTableNameConvention>();`.

I've since moved on to EFCore and needed to map the entities to the correct table names. Rather than doing each one individually I instead used the following:

```
foreach (var entity in modelBuilder.Model.GetEntityTypes())
            {
                entity.Relational().TableName = entity.DisplayName();
            }
```

Whilst this worked originally, it seems to have an issue with `owned` entities. I've since mapped each individually and is working ok.
